src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -110,6 +126,101 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
     });
   }
 
+  get canChangeDatasetOrientation(): boolean {
+    const sheetNames = new Set<string>();
+    const datasetZones: Zone[] = [];
+    const currentSheetName = this.env.model.getters.getActiveSheet().name;
+    for (const dataSet of this.dataSets) {
+      if (!isXcRepresentation(dataSet.dataRange)) {
+        return false;
+      }
+      const reference = splitReference(dataSet.dataRange);
+      const zone = toUnboundedZone(reference.xc);
+      if (zone.bottom === undefined || zone.right === undefined) {
+        return false;
+      }
+      datasetZones.push(zone as Zone);
+      sheetNames.add(reference.sheetName || currentSheetName);
+    }
+    if (this.labelRange) {
+      if (!isXcRepresentation(this.labelRange)) {
+        return false;
+      }
+      const labelReference = splitReference(this.labelRange);
+      const labelZone = toUnboundedZone(labelReference.xc);
+      if (labelZone.bottom === undefined || labelZone.right === undefined) {
+        return false;
+      }
+      datasetZones.push(labelZone as Zone);
+      sheetNames.add(labelReference.sheetName || currentSheetName);
+    }
+    if (sheetNames.size > 1) {
```

Comment:
We could avoid some iterations if we move this check just after the `sheetNames.add(reference.sheetName || currentSheetName);`